core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -561,7 +555,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         .last(newLastScanTime - 1)
         .asScala
         .toList
-      stale.foreach { log =>
+      stale.filterNot(isProcessing(_)).foreach { log =>
```

Comment:
`filterNot(isProcessing)`

But there's a problem here. Let's say you're unlucky and the cleaner task gets here for a log that should be cleanup up, but for some reason is still being processed by the tasks created by "checkForLogs".

Instead of cleaning that log, it will instead be skipped until the next time the cleaner task run, which may be quote a long time later.

Reply:
yes, I think we do not need filter these logs which is contained in `processing`.

The scene that log is both in `listing` and `processing` would only happen when  a log has been processed and written to the `listing` but the finally block(remove it from `processing`)
 has not been executed.

But for this case, even if we cleanup these logs, it would has no impaction.
So, I think we don't need filter these logs here.
@vanzin

Reply:
Is it safe if "cleaner task" and "replay task" access the same log directory concurrently? I guess we still need to filter these logs for safety, but we need to also retry cleaning up these logs sooner, at least sooner than the interval of cleaner.

Reply:
Sorry for late reply, I was on my National Day holiday for past several days.

> Instead of cleaning that log, it will instead be skipped until the next time the cleaner task run, which may be quote a long time later.

>  but we need to also retry cleaning up these logs sooner, at least sooner than the interval of cleaner.

It sound a little complex, should we clean these logs with half of  interval of cleaner later.

Reply:
@turboFei 
Sorry I got lost on this and revisit this now.
I agree it could be complicated, as it's really just a cleaner. For workaround, we can just do two iterations in a single call, on first iteration try to do all, on second iteration try only with skipped logs. 
@vanzin Would the workaround work for you? Or would we have to reschedule the interval?

Reply:
I don't think that solves the problem. To solve that problem, the task that's processing the logs should check at the end whether the log file has expired.

Reply:
Thanks, make sense.